apps/arena/lib/arena/game/skill.ex

```diff
@@ -258,6 +180,51 @@ defmodule Arena.Game.Skill do
     end)
   end
 
+  def do_mechanic(
+        game_state,
+        entity,
+        %{type: "simple_shoot"} = simple_shoot,
+        %{skill_direction: skill_direction, can_pick_destination: true} = skill_params
+      ) do
+    last_id = game_state.last_id + 1
+    entity_player_owner = get_entity_player_owner(game_state, entity)
+
+    direction = maybe_multiply_by_range(skill_direction, skill_params.auto_aim?, simple_shoot.range)
+
+    target_position = %{
+      x: entity.position.x + direction.x,
+      y: entity.position.y + direction.y
+    }
+
+    direction =
+      Utils.normalize(%{
+        x: target_position.x - entity.position.x,
+        y: target_position.y - entity.position.y
+      })
+
+    duration_ms = Physics.calculate_duration(entity.position, target_position, simple_shoot.speed, simple_shoot.range)
```

Comment:
What are the cases where the target_position is outside the range? Shouldn't we handle or restrict that before doing the calculation in the physics module? I think we're overfitting our physics to handle backend's unhandled edge cases, but maybe I'm wrong.

Reply:
Now projectiles can pick a destination, maybe the client cheats and the destination has more range than the possible one provided by the mechanic.
Also this happens when the auto-aim have more range than the projectile range